### PR TITLE
fix undefined content with embeds, add support for ban reasons

### PR DESCRIPTION
**Pull request by briantanner**
Fixes undefined content when sending embeds.
Adds support for ban reasons.

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -882,13 +882,13 @@ class Client extends EventEmitter {
             content = {
                 content: "" + content
             };
-        } else if(content && typeof content.content !== "string") {
+        } else if(content && content.content && typeof content.content !== "string") {
```
Comment (abalabahaha) [inline lib/Client.js]: content.content !== undefined

File: lib/Client.js [hunk 2]
```diff
@@ -1183,12 +1183,13 @@ class Client extends EventEmitter {
     * @arg {Number} [deleteMessageDays=0] Number of days to delete messages for, between 0-7 inclusive
     * @returns {Promise}
     */
-    banGuildMember(guildID, userID, deleteMessageDays) {
+    banGuildMember(guildID, userID, deleteMessageDays, reason) {
```
Comment (abalabahaha) [inline lib/Client.js]: Doc this pls

File: lib/Client.js [hunk 3]
```diff
@@ -882,13 +882,13 @@ class Client extends EventEmitter {
...
             };
-        } else if(content && typeof content.content !== "string") {
+        } else if(content && content.content && typeof content.content !== "string") {
             content.content = "" + content.content;
         }
         if(!content.content && !file && !content.embed) {
             return Promise.reject(new Error("No content, file, or embed"));
         }
-        if(content.disableEveryone !== undefined ? content.disableEveryone : this.options.disableEveryone) {
+        if(content.content && (content.disableEveryone !== undefined ? content.disableEveryone : this.options.disableEveryone)) {
```
Comment (abalabahaha) [inline lib/Client.js:891]: What's the point of this change?

**Comment by abalabahaha** [comment]
In the future please split bug and feature PRs

File: lib/Client.js (hunk 3 above)
Comment (briantanner) [reply lib/Client.js:891]: It was throwing an error replace of undefined.

**Pull request merged**